runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5275,6 +5275,11 @@ class INTERPRETER_CLASS
 		/* store the current Continuation state and swap to carrier thread stack */
 		yieldContinuation(_currentThread);
 
+		j9object_t continuationObject = J9VMJAVALANGTHREAD_CONT(_currentThread, _currentThread->carrierThreadObject);
+		if(J9VMJDKINTERNALVMCONTINUATION_FINISHED(_currentThread, continuationObject)) {
+			freeContinuation(_currentThread, continuationObject);
+		}
+
```

Comment:
it probably makes sense to be inside yield method
'finished' info is already fetched there
also code would be symmetric to how we create continuation (early in enterContinuation method)

Reply:
fixed